### PR TITLE
Handled exceptions should notify only once

### LINKED ISSUE #724: Fix broken test test_breaking_into_raised_exceptions_only
_From @DonJayamanne on August 1, 2018 5:23_

Currently skipped

_Copied from original issue: Microsoft/vscode-python#2292_

### PR DESCRIPTION
**Pull request by karthiknadig**
Fixes #724, #723 
Enables several other tests.

### CODE REVIEW

File: ptvsd/wrapper.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ def get_break_mode(self, name):
     def add_exception_break(self, exception, break_raised, break_uncaught,
                             skip_stdlib=False):
 
-        notify_on_handled_exceptions = 1 if break_raised else 0
+        notify_on_handled_exceptions = 2 if break_raised else 0
```
Comment (DonJayamanne) [inline ptvsd/wrapper.py]: Please use a constant for these magic numbers `1`, `0`, `2`
Previously i assumed these were booleans. Clearly these numbers have some special meaning.

File: tests/system_tests/test_web_frameworks.py [hunk 2]
```diff
@@ -258,6 +262,9 @@ def run_test_with_handled_exception(self, debug_info, framework,
             web_client_thread.start()
             web_client_thread.join(timeout=1)
 
+            # Give enough time for all events to be received
+            time.sleep(3)
```
Comment (DonJayamanne) [inline tests/system_tests/test_web_frameworks.py]: is there no way to query the events received and wait in a loop.
Might be better to do so, as I've done something similar in the past and have noticed such tests to fail occasionally (possibly when server is under load).

Comment (karthiknadig) [reply tests/system_tests/test_web_frameworks.py]: I will change how we handle this.

Comment (DonJayamanne) [reply tests/system_tests/test_web_frameworks.py]: Not mandatory, but good if we could

Comment (karthiknadig) [reply tests/system_tests/test_web_frameworks.py]: Created this #755 to tackle the problem with web frame work tests.

**Comment by karthiknadig** [comment]
@DonJayamanne I addressed the flaky system tests issue in #760 . Please approve this.

**Comment by karthiknadig** [comment]
Due to a change in VSTS there are multiple `ptvsd-testing-automation` listed. I will remote the required requirement until this is settled. This changed 2 times today:
image

**Comment by karthiknadig** [comment]
@DonJayamanne can you approve this?

**Comment by karthiknadig** [comment]
Waiting for decision on should we use old ptvsd behavior or C# behavior, by default, for raised exceptions. Another item is, should we add a setting (in ptvsd), which allows control of this behavior via `debugOptions`. It may be useful in rare cases. The thing to consider would be should this setting be exposed to VSC/VS users.
@DonJayamanne let me know what gets decided here.

**Comment by karthiknadig** [comment]
Closing this. We can re-visit this again later.

**Pull request closed without merging**